internal/app/machined/internal/phase/network/network.go

```diff
@@ -0,0 +1,39 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package network
+
+import (
+	"github.com/talos-systems/talos/internal/app/machined/internal/phase"
+	"github.com/talos-systems/talos/internal/app/machined/internal/platform"
+	"github.com/talos-systems/talos/internal/app/machined/internal/runtime"
+	"github.com/talos-systems/talos/internal/pkg/network"
+	"github.com/talos-systems/talos/pkg/userdata"
+)
+
+// UserDefinedNetwork represents the UserDefinedNetwork task.
+type UserDefinedNetwork struct{}
+
+// NewUserDefinedNetworkTask initializes and returns an UserDefinedNetwork task.
+func NewUserDefinedNetworkTask() phase.Task {
+	return &UserDefinedNetwork{}
+}
+
+// RuntimeFunc returns the runtime function.
+func (task *UserDefinedNetwork) RuntimeFunc(mode runtime.Mode) phase.RuntimeFunc {
+	switch mode {
+	case runtime.Standard:
+		return task.runtime
```

Comment:
feels odd that we don't do anything for the container here, but i guess it makes sense. we'd always see the DHCP logs/errors previously.

Reply:
Yeah setting up networking defined by the machine config in a container doesn't make sense so I left it out. So we return `nil` here to indicate to the phase runner that there is nothing to run.